crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -37,20 +37,32 @@ pub(crate) async fn farm(
 
     let reward_address = reward_address.unwrap_or(address);
 
+    info!("Connecting to node at {}", node_rpc_url);
+    let client = WsRpc::new(&node_rpc_url).await?;
+
+    let farmer_metadata = client
+        .farmer_metadata()
+        .await
+        .map_err(|error| anyhow::Error::msg(error.to_string()))?;
+
     // TODO: This doesn't account for the fact that node can
     // have a completely different history to what farmer expects
     info!("Opening plot");
     let plot_fut = tokio::task::spawn_blocking({
         let base_directory = base_directory.clone();
+        let plot_size = plot_size.map(|plot_size| plot_size / PIECE_SIZE as u64);
+
+        if let Some(plot_size) = plot_size {
```

Comment:
I'd probably at least leave a TODO here because technically this is incorrect right now. Yes, we print the warning, but we don't actually limit the max plot size. However, out plot size limit on `Plot` level should no longer be optional, it should always be limited by at least `MAX_PLOT_SIZE`, it doesn't make any sense to store anything outside of that range because it will fail at consensus level anyway.